soh/soh/OTRGlobals.cpp

```diff
@@ -870,7 +872,7 @@ OTRVersion ReadPortVersionFromOTR(std::string otrPath) {
     // Use a temporary archive instance to load the otr and read the version file
     auto archive = LUS::OtrArchive(otrPath);
     if (archive.Open()) {
-        auto t = archive.LoadFileRaw("portVersion");
+        auto t = archive.LoadFile("portVersion", std::make_shared<LUS::ResourceInitData>());
```

Comment:
this is the only SoH side change that isn't part of https://github.com/HarbourMasters/Shipwright/pull/3998

Reply:
So `LoadFileRaw` is essentially replaced by `LoadFile` with a non-null initdata?

Reply:
more or less, if you look at `LoadFile` in archive now <https://github.com/Kenix3/libultraship/blob/d70fd7de2315388c9a99742fd6c634b2cd8be138/src/resource/archive/Archive.cpp#L206-L241> you'll see

```cpp
    std::shared_ptr<File> fileToLoad = nullptr;

    if (initData != nullptr) {
        fileToLoad = LoadFileRaw(filePath);
        fileToLoad->InitData = initData;
    } else {
        //  [...]
    }

    return fileToLoad;
```

but using it directly still feels hacky.

i'm also not sure if kenix wants to move `LoadFile` itself away from being public. if that's the case i'll need to refactor `ReadPortVersionFromOTR` to use a temporary `ResourceManager` (which also feels hacky) and that will definitely take a bit more code

Reply:
If it gets to that point, then we could probably just re-think how `portVersion` is represented in the archive itself. As long as the update with it lands on a major/minor update then our version detection logic would enforce a regeneration anyways, so the user experience would be the same either way without us having to maintain a deprecated solution.

Reply:
> If it gets to that point, then we could probably just re-think how `portVersion` is represented in the archive itself.

for sure, it's really just a matter of figuring out how we want to handle needing to read from archives when we don't have a resource manager yet

Reply:
just asked kenix about `LoadFile`

> Public is fine, but you're encouraged to use *ResourceManager* instead of either ArchiveManager or Archive.

Reply:
so this is a reasonable way to handle this use case, if we think of a better one we can figure out how we want to implement it, but we definitely won't need to hack together a temporary resourcemanager based solution